odoo/addons/base/res/res_currency.py

```diff
@@ -169,34 +174,38 @@ def is_zero(self, amount):
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
     @api.model
-    def _get_conversion_rate(self, from_currency, to_currency):
+    def _get_conversion_rate(self, from_currency, to_currency, company_id, date):
         from_currency = from_currency.with_env(self.env)
         to_currency = to_currency.with_env(self.env)
-        return to_currency.rate / from_currency.rate
+        currency_rates = (from_currency + to_currency).get_rates(company_id, date)
+        return currency_rates.get(to_currency.id) / currency_rates.get(from_currency.id)
```

Comment:
Can probably use `currency_rates[]` here, if the currency is missing from the rates map for some reason it's going to blow up anyway (by trying to compute `None / 63` or some such nonsense) we may as well provide the ~~most helpful~~ least unhelpful error.